src/editor.cpp

```diff
@@ -0,0 +1,580 @@
+        return;
+    }
+    const std::vector<nlohmann::json>& allEEPROMs =
+        m_jsonFile["frus"][m_vpdFilePath]
+            .get_ref<const nlohmann::json::array_t&>();
+
+    types::ObjectMap allObjects;
+
+    for (const auto& thisEEPROM : allEEPROMs)
+    {
+        types::PropertyMap properties;
+        types::InterfaceMap interfaces;
+
+        // by default inherit property is true
+        bool isInherit = true;
+        if (thisEEPROM.contains("inherit"))
+            isInherit = thisEEPROM["inherit"].get<bool>();
+
+        if (isInherit)
+        {
+            properties.emplace(
+                utils::getDbusNameForThisKw(m_thisRecord.keywordName),
+                m_thisRecord.newData);
+            interfaces.emplace(
+                (std::string{"com.ibm.ipzvpd."} + m_thisRecord.recordName),
+                std::move(properties));
+            allObjects.emplace((thisEEPROM["inventoryPath"].get<std::string>()),
+                               std::move(interfaces));
+
+            updateCI(thisEEPROM["inventoryPath"]
+                         .get_ref<const nlohmann::json::string_t&>());
+        }
+        updateEI(thisEEPROM, thisEEPROM["inventoryPath"]
+                                 .get_ref<const nlohmann::json::string_t&>());
+
+        if (thisEEPROM.contains("copyRecords"))
+        {
+            if (find(thisEEPROM["copyRecords"].begin(),
+                     thisEEPROM["copyRecords"].end(),
+                     m_thisRecord.recordName) !=
+                thisEEPROM["copyRecords"].end())
+            {
+                properties.emplace(m_thisRecord.keywordName,
+                                   m_thisRecord.newData);
+                interfaces.emplace(
+                    (std::string{"com.ibm.ipzvpd."} + m_thisRecord.recordName),
+                    std::move(properties));
+                allObjects.emplace(
+                    (thisEEPROM["inventoryPath"].get<std::string>()),
+                    std::move(interfaces));
+            }
+        }
+    }
+    // Notify PIM
+    utils::callPIM(std::move(allObjects));
+ std::cout<<"DBG:return updateCache \n";
+}
+
+void vpdEditor::modifyVpd(const bool& isCacheUpdateRequired)
+{
+    std::cout << "DBG Entered Modify vpd:\n";
+
+    try
+    {
+        // To avoid any data corruption,restrict the BMC from rebooting when VPD
+        // is being written.
+        enableRebootGuard();
+
+        // Read VPD offset if applicable.
+        if (!m_jsonFile.empty())
+        {
+            m_vpdStartOffset = utils::getVPDOffset(m_jsonFile, m_vpdFilePath);
+        }
+
+        m_vpdFileStream.exceptions(std::ifstream::badbit |
+                                   std::ifstream::failbit);
+
+        utils::getVpdDataInVector(m_vpdFileStream, m_vpdFilePath, m_vpdBytes,
+                                  m_vpdStartOffset);
+
+        if (!m_vpdBytes.empty())
+        {
+            // Validate Header part
+            ValidateHeader();
```

Comment:
Should we need caps in start of function name?

Reply:
No